src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/PreferJaggedArraysOverMultidimensional.cs

```diff
@@ -76,33 +76,37 @@ private static void AnalyzeProperty(SymbolAnalysisContext context)
         {
             var property = (IPropertySymbol)context.Symbol;
 
-            // If its an override then don't report it as it can only be fixed in the base type.
-            if (!property.IsOverride)
+            // Bail-out when override or interface implementation as it can only be fixed in the base type.
+            if (property.IsOverride || property.IsImplementationOfAnyInterfaceMember())
```

Comment:
`IsImplementationOfAnyInterfaceMember` is an expensive check. Can we switch it to be done after `IsMultiDimensionalArray` check?

Reply:
I am not sure the code will look nice because we want this check to be done whatever the result of `IsMultiDimensionalArray` is.

```csharp
if (property.IsOverride) return;

if (IsMutltiDim(property.Type) && !property.IsImpl()) report;

if (!property.IsImpl()) AnalyzeParams();
```

Reply:
You can refactor `AnalyzeParams` itself takes the property and has `if (IsMutltiDim(parameter.Type) && !containingProperty.IsImpl()) report;`

Reply:
So we end-up doing twice the check which is expensive. I'll store the result in a variable and pass-it along.

Reply:
I will push a commit with the change you are suggesting but it seems a lot slower as the expensive call is now done multiple times...